lms/djangoapps/courseware/module_render.py

```diff
@@ -335,6 +337,38 @@ def get_module(user, request, usage_key, field_data_cache,
         return None
 
 
+def display_access_messages(user, block, view, frag, context):
+    blocked_prior_sibling = RequestCache('display_access_messages_prior_sibling')
+
+    load_access = has_access(user, 'load', block, block.scope_ids.usage_id.course_key)
+    if load_access:
+        blocked_prior_sibling.delete(block.parent)
+        return frag
+
+    prior_sibling = blocked_prior_sibling.get_cached_response(block.parent)
+
+    if prior_sibling.is_found and prior_sibling.value == load_access:
+        return Fragment(u"")
+    else:
+        blocked_prior_sibling.set(block.parent, load_access)
+
+    if load_access.user_fragment:
+        msg_fragment = load_access.user_fragment
+    elif load_access.user_message:
+        msg_fragment = Fragment(textwrap.dedent(u"""\
+            <div>{}</div>
+        """.format(load_access.user_message)))
+    else:
+        msg_fragment = Fragment(u"")
+
+    if load_access.developer_message and has_access(user, 'staff', block, block.scope_ids.usage_id.course_key):
```

Comment:
does staff include course staff or just superusers?  
I don't know if we want course staff to see the developer message

Reply:
I believe this does not include course staff.